test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -283,3 +292,84 @@
 
             ;; we should have three instances with the constant and one without.
             (is (true? (check-jrubies-for-constant-counts pool-context 3 1)))))))))
+
+(defprotocol BonusService
+  (bonus-service-fn [this]))
+
+(deftest ^:integration test-hup-comes-back
+  (testing "After a HUP signal puppetserver can still handle requests"
+    (let [call-seq (atom [])
+          lc-fn (fn [context action] (swap! call-seq conj action) context)
+          bonus-service (tk-services/service BonusService
+                          [[:MasterService]]
+                          (init [this context] (lc-fn context :init-bonus-service))
+                          (start [this context] (lc-fn context :start-bonus-service))
+                          (stop [this context] (lc-fn context :stop-bonus-service))
+                          (bonus-service-fn [this] (lc-fn nil :bonus-service-fn)))]
+      (bootstrap/with-puppetserver-running-with-services
+       app
+       (conj (tk-bootstrap/parse-bootstrap-config! bootstrap/dev-bootstrap-file) bonus-service)
+       {:jruby-puppet {:max-active-instances 1}}
+       (tk-internal/register-sighup-handler [app])
+       (beckon/raise! "HUP")
+       (let [start (System/currentTimeMillis)]
+         (while (and (not= (count @call-seq) 5)
+                     (< (- (System/currentTimeMillis) start) 90000))
+           (Thread/yield)))
+       ; 5 elements in the call-seq corresponds to init, start, stop, init, start for the BonusService
+       (is (= (count @call-seq) 5))
```

Comment:
Any reason to not just assert `(is (= @call-seq [:init-bonus-service :start- ...]))` ?